### PR TITLE
Model Visits

### PR DESCRIPTION
**Pull request by rpiaggio**
Events and dataset information not modeled yet.

I'm not sure we gain much by having a `Step` trait that groups `FutureStep` and `StepRecord`. In fact, I purposely didn't define a `Eq[Step]` instance.

Opinions?

### CODE REVIEW

File: modules/core/shared/src/main/scala/lucuma/core/model/sequence/Step.scala [hunk 1]
```diff
@@ -86,17 +95,117 @@ object Step extends WithUid('s') {
...
+
+object StepRecord {
+  final case class GmosNorth(
+    id:               Step.Id,
+    created:          Instant,
+    startTime:        Option[Instant],
+    endTime:          Option[Instant],
+    instrumentConfig: DynamicConfig.GmosNorth,
+    stepConfig:       StepConfig
+  ) extends StepRecord
```
Comment (swalker2m) [inline modules/core/shared/src/main/scala/lucuma/core/model/sequence/Step.scala:130]: I'm not sure it matters for the purpose of modeling this, but as I mentioned in a separate email the times come from (and must be consistent with) step events from Observe.

File: modules/core/shared/src/main/scala/lucuma/core/model/sequence/Visit.scala [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import java.time.Instant
+
+sealed trait Visit {
+  def id: Visit.Id
+  def created: Instant
+  def staticConfig: StaticConfig
+  def steps: List[StepRecord]
+
+  def startTime: Option[Instant] = steps.headOption.flatMap(_.startTime)
+  def endTime: Option[Instant]   = steps.lastOption.flatMap(_.endTime)
```
Comment (swalker2m) [inline modules/core/shared/src/main/scala/lucuma/core/model/sequence/Visit.scala:26]: These times will come from sequence events, when present (which will be the usual case).

File: modules/core/shared/src/main/scala/lucuma/core/model/sequence/Step.scala (hunk 1 above)
Comment (rpiaggio) [reply modules/core/shared/src/main/scala/lucuma/core/model/sequence/Step.scala:130]: I see. I'll add the step events then and remove the start and end times.

File: modules/core/shared/src/main/scala/lucuma/core/model/sequence/Visit.scala (hunk 2 above)
Comment (rpiaggio) [reply modules/core/shared/src/main/scala/lucuma/core/model/sequence/Visit.scala:26]: Same here.

**Comment by rpiaggio** [comment]
Closing this, the new classes will go to `lucuma-schemas`.

**Pull request closed without merging**